atlas-core/src/main/java/io/atlasmap/core/BaseModuleValidationService.java

```diff
@@ -76,9 +76,41 @@ public String value() {
 
     protected abstract AtlasModuleDetail getModuleDetail();
 
+    protected void validateCombineMapping(Mapping mapping, List<Validation> validations) {
```

Comment:
@jpav can you move this method next to/right before the validateSeparateMapping() or validateMapMapping()?

Reply:
Well, this ends up being a religious issue...;-) From lots of experience, I long ago came to the conclusion that trying to organize methods according to their relevance to other methods ultimately fails.  It's too easy for methods to have multiple relationships to several other methods, causing this "proximity" technique to fall apart.  I instead try to keep all methods in alphabetical order, which, AFAICT, has not significant downfall.

Reply:
At least it looks weird to have this above the validateMapping() which is publicly exposed, it's not reader friendly. We're not strictly following the order of public>protected>private, but it often gains readability.

Reply:
And also I'd like to try to avoid breaking intention of original author without practical reason.

Reply:
The practical reason is exactly for readability, plus we're the new authors ;-).  *My* intention is to move towards an alphabetical organization, but of course that won't fly if we're not on the same page.  Do you dislike alphabetical organization?

Reply:
Alphabetical order with disregard of access modifier and dependency looks just odd to me. I haven't ever seen that policy in other open source projects I have ever looked, Apache Camel, WildFly, JBossESB, SwitchYard, HornetQ, ActiveMQ, Narayana... I think people have looked at those would be surprised with it like me, which is not so good for readability for those people. And I don't see anything following that policy in the atlasmap project, which means it's not what project founder intended.
I guess alphabetical order looks good only for relatively less complex collections, like a bunch of one-line definitions of constant or XML entries.

Reply:
BTW do you know the old Java coding conventions have public>protected>private order for the methods? Recently we're not strictly following that, but it's still effective as a principal.

Reply:
So I don't care the order among other subroutines at all, but do care if one of them is above the public validateMapping().

Reply:
Well, I'll make one last comment, but I'll go ahead and move the method to avoid belaboring this too long. Keep in mind the rest of this code-base doesn't follow that standard consistently at all. My opinion is still that alphabetical is the better way to go, regardless of the number of projects that don't follow this. If you look at any of those projects, you'll find their organization methods break down somewhere along the line. Also consider most IDEs, including Eclipse, either automatically sort things like projects, classes, methods, etc., alphabetically or have a button that does so. JavaDocs are also always alphabetical. Anyway, moving on...

Reply:
Strictly following single rule is not my interest, current code is far easier to read than alphabetical order for me. Note that you're proposing to change a fundamental rule on already established project.

Reply:
k, like I said, I'll change it for you. I respect that we just have a difference of opinion here. But you realize _**nothing**_ in this entire project is following any type of "fundamental rule" or has any "established" convention in this regard. This class alone fails to follow any type of convention before my additional method: 

- There's an enum sitting right in between some of the methods
- The getModuleDetail() method is protected but already sits between two public methods
- Methods don't consistently exist with related methods, such as the getFieldType() method residing between validation methods.

Most of the other classes are even more poorly organized that this one. Thus I fail to see my approach can possibly be "harmful" in any way to other members.

Having said that, going forward, are you okay with moving towards organizing members by visibility (public, protected, etc.)?

Reply:
At first, I'm sorry if you feel bad with my poor English...

> But you realize nothing in this entire project is following any type of "fundamental rule" or has any "established" convention in this regard. 

disagreed, a rule `public method goes above its subroutines` is obviously an established rule and it did helped me to fix nearly hundred of bugs in this project. First enter into the file from the public method found from stacktrace, and goes down when I need to look into subroutine.

> There's an enum sitting right in between some of the methods

Is this a problem?

> The getModuleDetail() method is protected but already sits between two public methods

I'm OK with this as abstract method is something author wanted to expose. But maybe a bit better move it under public.

> Methods don't consistently exist with related methods, such as the getFieldType() method residing between validation methods.

I guess this is here because validateField() right above invokes it. I don't think it's a problem. I'm fine getting protected abstract together right under public ones, or keep them under invokers. Trivial thing for me...

> Thus I fail to see how my approach can possibly be "harmful" in any way to other members.

Are you saying I'm not a member of this project? I said I felt harmful, it's annoying when debugging.

Let's say there are two rules `DO NOT KILL PEOPLE` and `DO NOT TELL A LIE`. Re-sorting in alphabetical order just sounds like `OK so much people are telling lies, so I made a new rule, I can kill any criminals.` for me.

Reply:
Wow, okay, I guess maybe I'm misinterpreting what you were saying.  I didn't realize you meant you were the one being harmed by this. I've been looking at this as merely a preference, not a major hinderance.  I apologize.  I'll concede the debate.